components-compose/src/main/java/uk/gov/hmrc/components/compose/molecule/selectrow/SelectRowView.kt

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.hmrc.components.compose.molecule.selectrow
+
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.offset
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.selection.selectable
+import androidx.compose.material.ripple.rememberRipple
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconToggleButton
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.semantics.Role
+import uk.gov.hmrc.components.compose.R
+import uk.gov.hmrc.components.compose.ui.theme.HmrcTheme
+
+object SelectRowView {
+
+    /**
+     * Composable function that creates a SelectRowView with icons and text.
+     *
+     * @param selectRowViewItems List of items to be displayed in the selectable rows.
+     * @param checkedIcon Resource ID for the checked icon.
+     * @param uncheckedIcon Resource ID for the unchecked icon.
+     * @param rowSelectedPosition Initial selection position of a SelectRowView
+     * @param errorText Error text to be displayed, if any.
+     * @param onRowSelected Callback function triggered when an item is clicked.
+     */
+    @Composable
+    operator fun invoke(
+        modifier: Modifier = Modifier,
+        selectRowViewItems: List<String>,
+        checkedIcon: Int = R.drawable.components_select_row_circle_checked,
+        uncheckedIcon: Int = R.drawable.components_select_row_circle_unchecked,
+        rowSelectedPosition: Int,
```

Comment:
I think `rowSelectedPosition` either needs to be a nullable parameter, or we should reintroduce the default parameter with the `const val NO_ROW_SELECTED_POSITION = -1`. 

My previous comment about these consts was specifically about not needing the `FIRST_ROW_SELECTED_POSITION` const for the first value, as it didn't really make sense to define that position explicitly but none of the others.. I don't think we need to provide consts for the items, but a default state for nothing selected seems useful. There will be common cases where we don't have anything pre-selected for the user.